front/lib/resources/vault_resource.ts

```diff
@@ -161,9 +161,7 @@ export class VaultResource extends BaseResource<VaultModel> {
   ): Promise<VaultResource[]> {
     const vaults = await this.baseFetch(auth);
     return vaults.filter(
-      (vault) =>
-        (auth.isAdmin() && adminsBypassACL) ||
-        auth.hasPermission([vault.acl()], "read")
+      (vault) => (auth.isAdmin() && adminsBypassACL) || vault.canRead(auth)
```

Comment:
can we move the `(auth.isAdmin() && adminsBypassACL)` logic inside `canRead` ?

Reply:
This has been changed by @Fraggle in https://github.com/dust-tt/dust/pull/7194. Will rebase.